tests/unit/test_endpoint_provider.py

```diff
@@ -544,3 +553,141 @@ def test_deeply_nested(self):
             ]
         }
         self.assertDictEqual(expected_rules_file, rules_file_dict_read)
+
+
+def provider_with_rules_factory(rules: Dict[str, str]):
+    """Class factory to create a provider charm class with the given rules.
+
+    Args:
+        rules: a mapping between filename and contents.
+    """
+
+    class ProviderCharm(CharmBase):
+        metadata_yaml = textwrap.dedent(
+            """
+            provides:
+              metrics-endpoint:
+                interface: prometheus_scrape
+            """
+        )
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args)
+
+            self.sandbox = TempFolderSandbox()
+            alert_rules_path = os.path.join(self.sandbox.root, "alert_rules")
+            for filepath, contents in rules.items():
+                self.sandbox.put_file(os.path.join(alert_rules_path, filepath), contents)
+
+            self.provider = MetricsEndpointProvider(
+                self, jobs=JOBS, alert_rules_path=alert_rules_path
+            )
+
+    return ProviderCharm
+
+
+class TestAlertRulesDoNotContainUnitNumber(unittest.TestCase):
+    """Feature: Consumer lib must not add a unit number to rule filters or labels.
+
+    Background: In lieu of SDI or pydantic, schema verification is done in unit tests.
+    """
+
+    def setUp(self):
+        pass
+
+    @patch_network_get(private_address="192.0.8.2")
+    def test_alert_rules_with_topology(self):
```

Comment:
Why do we need a separate test case here. There is already a test case to check that each alert rule is topology labeled `test_each_alert_rule_is_topology_labeled()`. Could this existing test case be extended to add two additional checks
```
      self.assertNotIn("juju_unit", rule["labels"])
      self.assertNotIn("juju_unit=", rule["expr"])
```

Reply:
Would you rather add two extra lines of code or add 2 dozen lines of code ? I see no issue with changing the name. Even the name as it is should be fine. The current name is just say that this test checks alert rules are labelled with topology. It does not say this test only checks alert rule labels are topology labelled.